src/audio/tdfb/tdfb.c

```diff
@@ -800,18 +769,12 @@ static int tdfb_prepare(struct comp_dev *dev)
 	memset(cd->in, 0, TDFB_IN_BUF_LENGTH * sizeof(int32_t));
 	memset(cd->out, 0, TDFB_IN_BUF_LENGTH * sizeof(int32_t));
 
-	ret = set_func(dev, source_c->stream.frame_fmt);
+	ret = set_func(mod, frame_fmt);
 	if (ret)
 		goto out;
 
-	/* The max. amount of processing need to be limited for sound direction
-	 * processing. Max frames is used in tdfb_direction_init() and copy().
-	 */
-	cd->max_frames = Q_MULTSR_16X16((int32_t)dev->frames, TDFB_MAX_FRAMES_MULT_Q14, 0, 14, 0);
```

Comment:
cd->max_frames can't be removed. As zero it will cause a too small buffer allocation and memory violation.

Reply:
Fixed now.